bundles/org.openhab.ui/web/src/components/config/config-sheet.vue

```diff
@@ -94,7 +94,7 @@ export default {
       this.showAdvanced = !this.showAdvanced // event.target.checked
     },
     updateParameter (parameter, value) {
-      if (!value) {
+      if (isNaN(value) || value === '') {
         // deleting the parameter sometimes lead to saves not updating it, so set it explicitely to null
```

Comment:
Perhaps some more explanation is needed here.

When the user clears the input, the natural thing to do would be to remove the config parameter from the resulting config object, but there's a catch: I noticed at least for PUT /rest/services/<serviceId>/config, if you don't specify a parameter, it won't be unset. For example if you first do a PUT of:
`{ "param1": "A", "param2": "B" }`
then:
`{ "param2": "C" }`
the actual config will be:
`{ "param1": "A", "param2": "C" }`

However if the second time you PUT:
`{ "param1": null, "param2": "C" }`
when you GET the config again you will have:
`{ "param2": "C" }`
as expected - "param1" has been unset by setting it to null.

So here if the input is blank (empty string for `<input type="text">` or NaN for `<input type="number">`) it's set to null to unset it from the config when the API is called. Hope it makes sense.

Reply:
I'm not sure I understand this yet - I need to digest it a bit more, but I guess this is linked to the issue I just added below? Sorry that I didn't see this till after adding that comment.

Reply:
When you think about it - it should probably fixed in core. If you use a HTTP PUT to update a config it should replace it completely, not merge it with the existing (that would actually be a PATCH - https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/PATCH)